include/zephyr/net/coap_mgmt.h

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2023 Basalte bv
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief CoAP Events code public header
+ */
+
+#ifndef ZEPHYR_INCLUDE_NET_COAP_MGMT_H_
+#define ZEPHYR_INCLUDE_NET_COAP_MGMT_H_
+
+#include <zephyr/net/net_mgmt.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief CoAP Manager Events
+ * @defgroup coap_mgmt CoAP Manager Events
+ * @ingroup networking
+ * @{
+ */
+
+/** @cond INTERNAL_HIDDEN */
+
+/* CoAP events */
+#define _NET_COAP_LAYER		NET_MGMT_LAYER_L4
+#define _NET_COAP_CODE		0x1c0
+#define _NET_COAP_IF_BASE	(NET_MGMT_EVENT_BIT |			\
+				 NET_MGMT_LAYER(_NET_COAP_LAYER) |	\
+				 NET_MGMT_LAYER_CODE(_NET_COAP_CODE))
+
+/** @endcond */
+
+enum net_event_coap_cmd {
+	NET_EVENT_COAP_CMD_SERVICE_STARTED = 1,
+	NET_EVENT_COAP_CMD_SERVICE_STOPPED,
+};
+
+/**
+ * @brief coap_mgmt event raised when a service has started
+ */
+#define NET_EVENT_COAP_SERVICE_STARTED			\
+	(_NET_COAP_IF_BASE | NET_EVENT_COAP_CMD_SERVICE_STARTED)
```

Comment:
Is this only for events, or should I also add start and stop commands to be used with `CONFIG_NET_MGMT`?

Reply:
`NET_EVENT_*_CMD_*` could be also used to regsister `NET_REQUEST_*` and the request handler, I'm not sure if it's mandatory though. IMO in this case it's an unecessary overhead, the apps are better to use CoAP APIs directly and just stick to the events layer. Unless there's some usecase for `net_mgmt()` I don't see here.